js/client.js

```diff
@@ -1114,20 +1114,31 @@
 						}
 					}
 				} else {
+					var name = formatsList[j];
 					var searchShow = true;
 					var challengeShow = true;
 					var team = null;
-					var name = formatsList[j];
-					if (name.substr(name.length - 2) === ',#') { // preset teams
-						team = 'preset';
-						name = name.substr(0, name.length - 2);
-					}
-					if (name.substr(name.length - 2) === ',,') { // search-only
-						challengeShow = false;
-						name = name.substr(0, name.length - 2);
-					} else if (name.substr(name.length - 1) === ',') { // challenge-only
-						searchShow = false;
-						name = name.substr(0, name.length - 1);
+					var lastCommaIndex = name.lastIndexOf(',');
+					var code = lastCommaIndex >= 0 ? parseInt(name.substr(lastCommaIndex + 1), 16) : NaN;
+					if (!isNaN(code)) {
+						name = name.substr(0, lastCommaIndex);
+						if (code & 1) team = 'preset';
+						if (!(code & 2)) searchShow = false;
+						if (!(code & 4)) challengeShow = false;
+						if (!(code & 8)) tournamentShow = false;
+					} else {
+						/* Backwards compatibility: late 0.9.0 -> early 0.10.0 */
```

Comment:
> early 0.10.0

Just increment the version number to 0.10.1 if you're going to change the protocol.